CRUD operations for users

shernanm opened the pull request:
## Summary
- CRUD operations for users created.
- Custom exceptions added.
## Trello Card
- [Trello Card](https://trello.com/c/wTE0YkQC)

Commit:
CRUD operations for users created and custom exceptions added

leandrocostawolox commented on src/main/java/wolox/training/controllers/UserController.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @GetMapping
+    public Iterable findAll() {
```
You can return here an `Iterable<User>` to be more specific

leandrocostawolox commented on src/main/java/wolox/training/controllers/UserController.java:
```diff
@@ -0,0 +1,78 @@
+        return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+    }
+
+    @PostMapping("{user_id}/book/{book_id}")
```
I think in this endpoint we could use plural, I mean: `/books/` instead of `/book/`

leandrocostawolox commented on src/main/java/wolox/training/controllers/UserController.java:
```diff
@@ -0,0 +1,78 @@
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("{user_id}/book/{book_id}")
```
I think in this endpoint we could use plural, I mean: `/books/` instead of `/book/`

leandrocostawolox commented on src/main/java/wolox/training/exceptions/UserIdMismatchException.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.BAD_REQUEST, reason = "User id wrong")
```
Just a minor text fix, the reason should be: `Wrong user id`

leandrocostawolox reviewed (changes requested):
Good job! Check my comments :)

Commit:
Changes requested in the PR

- The type of iterable returned when you want to get users or books was specified.
- The name of the endpoints was changed to plural. These endpoints are those that allow you to add or remove a book to the collection.
- Lastly, a small change was made in the reason for the exceptions when an incorrect user or book id was sent.

Commit:
Merge branch 'add-user-repository' into user-CRUD-operations
